### PR TITLE
Eslint migration

### LINKED ISSUE #66: Migrate from TSLint to ESLint
Update system linter to use the newest tslint integration: https://github.com/typescript-eslint/typescript-eslint

### PR DESCRIPTION
**Pull request by Allcharles**
# Upgraded from TSLint to ESLint

Upgraded from TSLint which has been deprecated for the past year.

## Issues

Fixes #66 

## Final Checklist

- [x] Assign reviewers if you have permission
- [x] Assign labels if you have permission
- [x] Link issues related to PR
- [x] Ensure project linter is not producing any warnings (`npm run lint`)
- [x] Ensure build is passing on all browsers (`npm run test:all`)
- [x] Ensure CI build is passing

### CODE REVIEW

**Comment by Allcharles** [comment]
Still need to address some issues with ESLint hitting and OOM problem. Will solve by extending the RAM limit of node

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      "extends": [
+        "plugin:@angular-eslint/ng-cli-compat",
+        "plugin:@angular-eslint/ng-cli-compat--formatting-add-on",
+        "plugin:@angular-eslint/template/process-inline-templates"
+      ],
+      "plugins": ["eslint-plugin-rxjs", "rxjs-angular"],
+      "rules": {
+        "no-underscore-dangle": "off",
+        "@typescript-eslint/member-ordering": "off",
+        // TODO Re-enable
```
Comment (atruskie) [inline .eslintrc.json]: link to issue?

**Review by atruskie — CHANGES_REQUESTED** [review]
all those naming convention disables add noise.

might as well make this a cleanup PR and get them all done

**Comment by atruskie** [comment]
Squash this before merging

**Review by atruskie — APPROVED** [review]

**Pull request merged**